AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSCosmosDb.h

```diff
@@ -0,0 +1,32 @@
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^MSCosmosDbCompletionHandler)(NSData *data, NSError *error);
+
+@class MSCosmosDbIngestion;
+@class MSTokenResult;
+
+@interface MSCosmosDb : NSObject
+
+/**
+ * Call CosmosDb Api.
+ *
+ * @param httpIngestion Http client.
+ * @param tokenResult Token result.
+ * @param documentId Document Id.
+ * @param httpVerb Http verb.
+ * @param body Http body.
+ * @param completion Completion callback.
+ *
+ */
++ (void)cosmosDbAsync:(MSCosmosDbIngestion *)httpIngestion
```

Comment:
it's called multiple times with different requests(defined by verb and body)

Reply:
@ElektrojungeMS any thoughts on naming?

Reply:
Following the [Cocoa Coding Guidelines](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html), I think the APi should start with a verb, and have the ingnestion param in the method name, something like:
```
+ (void)performCosmosDbAsyncOperationWithIngestion:(MSCosmosDbIngestion *)httpIngestion
          tokenResult:(MSTokenResult *)tokenResult
           documentId:(NSString *)documentId
             httpVerb:(NSString *)httpVerb
                 body:(NSString *)body
    completionHandler:(MSCosmosDbCompletionHandler)completionHandler;```

```
+ (void)performCosmosDbAsync:(MSCosmosDbIngestion *)httpIngestion
          tokenResult:(MSTokenResult *)tokenResult
           documentId:(NSString *)documentId
             httpVerb:(NSString *)httpVerb
                 body:(NSString *)body
    completionHandler:(MSCosmosDbCompletionHandler)completionHandler;
```
could work, too, although I’d be a bit confused because the ingestion is not part of the method signature.

and `httpIngestion` can be `ingestion` or `cosmosDbIngestion` to match the type.